src/redux/sagas/motions/createEditDomainMotion.ts

```diff
@@ -180,23 +193,67 @@ function* createEditDomainMotion({
     );
     yield takeFrom(createMotion.channel, ActionTypes.TRANSACTION_SUCCEEDED);
 
-    if (annotationMessage) {
-      const ipfsHash = yield call(uploadIfpsAnnotation, annotationMessage);
-      yield put(transactionPending(annotateMotion.id));
+    const apolloClient = getContext(ContextModule.ApolloClient);
+
+    if (isCreateDomain) {
+      /**
+       * Save domain metadata in the database
+       */
+      yield apolloClient.mutate<
+        CreateDomainMetadataMutation,
+        CreateDomainMetadataMutationVariables
+      >({
+        mutation: CreateDomainMetadataDocument,
+        variables: {
+          input: {
+            id: getMotionDomainDatabaseId(colonyAddress, txHash),
+            name: domainName,
+            color: domainColor || DomainColor.LightPink,
+            description: domainPurpose || '',
+          },
+        },
+      });
+    } else if (domain?.metadata) {
+      yield apolloClient.mutate<
+        CreateDomainMetadataMutation,
+        CreateDomainMetadataMutationVariables
+      >({
+        mutation: CreateDomainMetadataDocument,
+        variables: {
+          input: {
+            id: getMotionDomainDatabaseId(colonyAddress, txHash),
+            name: domainName,
+            color: domainColor || domain.metadata.color,
+            description: domainPurpose || domain.metadata.description,
+            changelog: getUpdatedDomainMetadataChangelog(
+              txHash,
+              domain.metadata,
+              domainName,
+              domainColor,
+              domainPurpose,
+            ),
+          },
+        },
+      });
+    }
+
+    // if (annotationMessage) {
+    //   const ipfsHash = yield call(uploadIfpsAnnotation, annotationMessage);
+    //   yield put(transactionPending(annotateMotion.id));
 
-      yield put(transactionAddParams(annotateMotion.id, [txHash, ipfsHash]));
+    //   yield put(transactionAddParams(annotateMotion.id, [txHash, ipfsHash]));
 
-      yield put(transactionReady(annotateMotion.id));
+    //   yield put(transactionReady(annotateMotion.id));
 
-      yield takeFrom(annotateMotion.channel, ActionTypes.TRANSACTION_SUCCEEDED);
-    }
+    //   yield takeFrom(annotateMotion.channel, ActionTypes.TRANSACTION_SUCCEEDED);
+    // }
     yield put<AllActions>({
       type: ActionTypes.MOTION_DOMAIN_CREATE_EDIT_SUCCESS,
       meta,
     });
 
     if (colonyName) {
-      yield routeRedirect(`/colony/${colonyName}/tx/${txHash}`, history);
+      yield routeRedirect(`/colony/${colonyName}/tx/${txHash}`, navigate);
```

Comment:
This should follow the example of the rootMotion so that we get auto-redirected to the motion page instead of needing a browser refresh.